lib/nostr.js

```diff
@@ -76,6 +112,12 @@ export class Relay {
     if (state !== WebSocket.CLOSING && state !== WebSocket.CLOSED) {
       this.ws.close()
     }
+    for (const sub of this.subs) {
+      if (sub.closed) continue
+      sub.reject(new Error('relay closed'))
+    }
+    this.subs = []
+    this._closed = true
```

Comment:
I believe you only need this to not run the reconnect logic on close? You could also simply remove the listener if you don't want it to get run, no need for `this._closed`.

_I also wonder if this line should be before `this.ws.close()` :thinking:_

Reply:
yes, you are right, it should go before this.ws.close().
I've also added a check inside the reconnection timeout, just in case it is set right before some code calls close().

this._closed is used in `get closed ` and `get connected` to make the relay look closed and disconnected even if the socket is still closing underneath.